torch/csrc/jit/codegen/cuda/codegen.cpp

```diff
@@ -59,23 +61,37 @@ class CudaKernelGenerator : private OptOutConstDispatch {
     }
 
     // Generate parameter declarations
-    for (Val* val : params) {
-      if (const auto tv = dynamic_cast<TensorView*>(val)) {
+    unsigned int duplicate_counter = 0;
+    for (auto i : c10::irange(params.size())) {
+      std::stringstream var_name_ss;
+      if (params[i]->isA<TensorView>()) {
+        var_name_ss << varName(params[i]->as<TensorView>());
+      } else {
+        var_name_ss << gen(params[i]);
+      }
+
+      // If value is duplicate in arguments just send a dummy val through
```

Comment:
The comment seems to suggest that we should `continue` in the true condition below.